File: algorithms/python/171_excel_sheet_column_number/excel_sheet_column_number.py
"""
171. excel sheet column number

https://leetcode.com/problems/excel-sheet-column-number/
https://leetcode-cn.com/problems/excel-sheet-column-number/

Excel 表列序号
"""


def title_to_number(column_title: str) -> int:
    """
    26 进制
    """
    if not column_title:
        return 0
    tmp = 1
    base = ord('A')
    ret = 0
    for c in column_title[::-1]:
        ret += (ord(c) - base + 1) * tmp
        tmp *= 26
    return ret

File: algorithms/171-excel-sheet-column-number/excel-sheet-column-number.go
package problem171

// titleToNumber 26进制
func titleToNumber(columnTitle string) int {
	if columnTitle == "" {
		return 0
	}
	tmp := 1
	ret := 0

	byteArr := []byte(columnTitle)
	base := int(byte('A'))
	for i := len(byteArr) - 1; i >= 0; i-- {
		ret += (int(byteArr[i]) - base + 1) * tmp
		tmp *= 26
	}

	return ret
}
